### PR TITLE
(GH-1564) Add env_var builtin plugin

### PR DESCRIPTION
**Pull request by beechtom**
This adds an `env_var` builtin plugin that looks up data from an
environment variable.

### CODE REVIEW

File: lib/bolt/plugin/env_var.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'concurrent/delay'
```
Comment (nicklewis) [inline lib/bolt/plugin/env_var.rb]: This is unused

Comment (beechtom) [reply lib/bolt/plugin/env_var.rb]: The perils of copy/paste. Updated.

Comment (nicklewis) [reply lib/bolt/plugin/env_var.rb]: The real joy is it's also unused in the place it was copy/pasted from :)

Comment (beechtom) [reply lib/bolt/plugin/env_var.rb]: I'll go ahead and remove it in #1568 then 👍

**Comment by nicklewis** [comment]
If we _did_ add an option to parse the value as JSON/YAML, I think this plugin could entirely supplant the `$BOLT_INVENTORY` feature.

**Pull request merged**